addons/website_event/controllers/main.py

```diff
@@ -38,31 +38,27 @@ def sdn(date):
 
         def sd(date):
             return fields.Datetime.to_string(date)
+
+        def get_month_filter_domain(filter_name, filter_string, first_day, months_delta):
+            return [filter_name, filter_string, [
+                ("date_end", ">=", sd(first_day + relativedelta(months=months_delta))),
+                ("date_begin", "<", (first_day + relativedelta(months=months_delta+1)).strftime(datetime_format))],
+                0]
+
         today = datetime.today()
+        first_day_of_the_month = today.replace(day=1)
+        datetime_format = '%Y-%m-%d 00:00:00'
         dates = [
-            ['all', _('Next Events'), [("date_end", ">", sd(today))], 0],
+            ['all', _('Upcoming Events'), [("date_end", ">", sd(today))], 0],
             ['today', _('Today'), [
                 ("date_end", ">", sd(today)),
                 ("date_begin", "<", sdn(today))],
                 0],
-            ['week', _('This Week'), [
-                ("date_end", ">=", sd(today + relativedelta(days=-today.weekday()))),
-                ("date_begin", "<", sdn(today + relativedelta(days=6-today.weekday())))],
-                0],
-            ['nextweek', _('Next Week'), [
-                ("date_end", ">=", sd(today + relativedelta(days=7-today.weekday()))),
-                ("date_begin", "<", sdn(today + relativedelta(days=13-today.weekday())))],
-                0],
-            ['month', _('This month'), [
-                ("date_end", ">=", sd(today.replace(day=1))),
-                ("date_begin", "<", (today.replace(day=1) + relativedelta(months=1)).strftime('%Y-%m-%d 00:00:00'))],
-                0],
-            ['nextmonth', _('Next month'), [
-                ("date_end", ">=", sd(today.replace(day=1) + relativedelta(months=1))),
-                ("date_begin", "<", (today.replace(day=1) + relativedelta(months=2)).strftime('%Y-%m-%d 00:00:00'))],
-                0],
+            get_month_filter_domain('month', _('This month'), first_day_of_the_month, 0),
+            get_month_filter_domain('nextmonth1', _((today + relativedelta(months=1)).strftime('%B')), first_day_of_the_month, 1),
```

Comment:
@mart-e We need your mighty localization knowledge on this matter.

Can you give the result of a function to the `_()` function?
e.g:
`_(a_march_date.strftime('%B'))`
`a_march_date.strftime('%B')` gives "March"

Will "March" be translated to "Mars" if the user has his locale set to french?

Reply:
Well I think the answer is in the guidelines:

```py
# bad: dynamic string, string concatenation, etc are forbidden!
# This does NOT work and messes up the translations!
error = _("'" + que_rec['question'] + "' \n")
```

Sorry for the useless ping!

Reply:
Indeed the `_` is not a magic google-translate like method, the translation must be present in the database in the first place. To do what you want, check babel instead.